testing/unittests/unittests.cpp

```diff
@@ -674,4 +674,136 @@ class Base58Test : public CppUnit::TestFixture
 CPPUNIT_TEST_SUITE_REGISTRATION( Base58Test );
 CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( Base58Test, "Base58Test" );
 
+thread_local unsigned temp = 0;  // Avoids clever compilers optimizing everything away
+static unsigned skip(unsigned j)
+{
+    temp += j;
+    return j+1;
+}
+static unsigned call_from_thread(unsigned count)
+{
+    unsigned tot = count;
+    for (int j = 0; j < count; j++)
+        tot += skip(j);
+    return tot;
+}
+
+class ThreadedPersistStressTest : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE( ThreadedPersistStressTest  );
+        CPPUNIT_TEST(testThreads);
+    CPPUNIT_TEST_SUITE_END();
+
+    void testThreads()
+    {
+        testThreadsX(0);
+        testThreadsX(1);
+        testThreadsX(2);
+        testThreadsX(3);
+    }
+    void testThreadsX(unsigned mode)
+    {
+        unsigned iters = 10000;
+        testThreadsXX(mode, 10, iters);
+        testThreadsXX(mode, 1000, iters);
+        testThreadsXX(mode, 2000, iters);
+        testThreadsXX(mode, 4000, iters);
+        testThreadsXX(mode, 8000, iters);
+        testThreadsXX(mode, 16000, iters);
+        testThreadsXX(mode, 32000, iters);
+        testThreadsXX(mode, 64000, iters);
+    }
+    void testThreadsXX(unsigned mode, unsigned count, unsigned iters)
+    {
+        unsigned start = msTick();
+        class Thread : public IThreaded
+        {
+        public:
+            Thread(unsigned _count) : count(_count) {}
+            virtual void threadmain() override
+            {
+                ret = call_from_thread(count);
+            }
+            unsigned count;
+            unsigned ret = 0;
+        } t1(count), t2(count), t3(count);
+        CThreadedPersistent thread1("1", &t1), thread2("2", &t2), thread3("3", &t3);
```

Comment:
Should be inside the switch, case 0, otherwise it distorts the timings for the rest